src/Shared/TempFileUtilities.cs

```diff
@@ -22,7 +22,7 @@ internal static partial class FileUtilities
         /// <param name="subfolder"></param>
         internal static string GetTemporaryDirectory(bool createDirectory = true, string subfolder = null)
         {
-            string temporaryDirectory = Path.Combine(Path.GetTempPath(), "Temporary" + Guid.NewGuid().ToString("N"), subfolder ?? string.Empty);
+            string temporaryDirectory = Path.Combine(Path.GetTempPath(), "MSBuild" + Guid.NewGuid().ToString("N"), subfolder ?? string.Empty);
```

Comment:
Should we do `"MSBuild", Guid..." so that MSBuild is its own directory, and Guids are subdirectories within it?

Reply:
Also modify any similar methods (if they exist)?

Reply:
Currently changing the name to MSBuild here works, but there are places where this method could be called instead of Path.GetTempDirectory (even within FileUtilities). Unfortunately this means changing a bunch of tests and I'd rather not piggy-back that onto this PR.

I say we take this change and I can make a new PR that refactors `TempFileUtilities.cs`.

If not, I can revert the change to using the MSBuild folder and we can merge this PR as is.

Thoughts?

Reply:
I just realized what you were asking for wasn't many `MSBuild<guid>` folders, but `temp/MSBuild/<all the guids folders>`. Given that this commit didn't actually help much, I'm going to revert it and _actually_ fix it in a different PR that's based off of this one.

Reply:
sounds great

Reply:
tracking this here: https://github.com/dotnet/msbuild/pull/6239